src/Tools/dotnet-trace/CommandLine/Commands/ConvertCommand.cs

```diff
@@ -12,27 +13,32 @@ internal static class ConvertCommandHandler
     {
         public static int ConvertFile(IConsole console, FileInfo inputFilename, TraceFileFormat format, FileInfo output)
         {
-            if ((int)format <= 0)
+            if (!Enum.IsDefined(format))
             {
-                Console.Error.WriteLine("--format is required.");
+                console.Error.WriteLine($"Please specify a valid option for the --format. Valid options are: {string.Join(", ", Enum.GetNames<TraceFileFormat>())}.");
                 return ErrorCodes.ArgumentError;
             }
 
-            if (format == TraceFileFormat.NetTrace)
+            string outputFilename = TraceFileFormatConverter.GetConvertedFilename(inputFilename.FullName, output?.FullName, format);
+
+            if (format != TraceFileFormat.NetTrace)
             {
-                Console.Error.WriteLine("Cannot convert a nettrace file to nettrace format.");
-                return ErrorCodes.ArgumentError;
+                TraceFileFormatConverter.ConvertToFormat(console, format, inputFilename.FullName, output.FullName);
             }
-
-            if (!inputFilename.Exists)
+            else
             {
-                Console.Error.WriteLine($"File '{inputFilename}' does not exist.");
-                return ErrorCodes.ArgumentError;
+                console.Out.WriteLine($"Copying nettrace to:\t{outputFilename}");
+                try
+                {
+                    File.Copy(inputFilename.FullName, outputFilename);
```

Comment:
Actually, is there a header differentiation between `.nettrace`, `.speedscope.json`, and `.chromium.json` that we can check?
Right now by allowing `--format NetTrace` this path allows users to "convert" non NetTrace files into NetTrace, misleading that we support `non-NetTrace -> NetTrace`
`dotnet-trace convert --format NetTrace my.speedscope.json` would be permitted in this case.

Unless we have a way to differentiate/check the input file format, I think we should remove the `--format NetTrace`.

Reply:
I'm happy with any option where the help text that says what is supported is consistent with what the tool actually supports. I'd prioritize purely based on whichever one is easier to implement and maintain.

Nettrace does have [defined header bytes](https://github.com/microsoft/perfview/blob/main/src/TraceEvent/EventPipe/EventPipeFormat.md#first-bytes-nettrace-magic) if you wanted to check them, but if updating the help text was easy then I could imagine it is simpler than adding validation logic and special case handling for identical input/output paths.

Reply:
sigh... turns out the help isn't as easy in this ancient version of System.Commandline. Even if I implement a HelpBuilder, I'd have to hook into parts and copy a lot of things because `Command` help writing [doesn't call into `Option` help parsing](https://github.com/dotnet/command-line-api/blob/08d0408ce80894e1c9801f05561fc7b8b583c390/src/System.CommandLine/Help/HelpBuilder.cs#L54-L67). I'd like to move off it, but Mike has tried, and it's hard given the dependency dump has on the parser. I'll implement header sniffing for now.